### ISSUE TITLE
Rule Proposal: Declaring structs and arrays before usage

### ISSUE THREAD

**Issue by KamasamaK** [cflint/CFLint#406]
The following are legal ways to declare a struct or array.
- `tokenizedAddress.streetName = "Main";`
- `sampleCollection[7] = "Hello, World";`

It should recommend being explicit with the type declaration in these cases. The thing that is really troublesome about this is the second example. Would you expect that to be a struct or array? Well, it actually depends on the CFML engine. ACF will make it an array, but Lucee will make it a struct.

**Comment by ryaneberly** [comment]
@KamasamaK 
 Can you provide a couple examples of potential false positives.  I added your two positives to cce8bcc  (the rule isn't implemented yet.

**Comment by KamasamaK** [comment]
Sure. These should not trigger the rule.

```
var addressComponents = {};
addressComponents.streetName = "Main";
```

```
var usStates = structNew("ordered", "text", "asc");
usStates["AL"] = "Alabama";
```

```
var sampleCollection = [];
sampleCollection[7] = "Hello, World";
```

```
var sampleCollection = arrayNew(1, false);
sampleCollection[1] = "Hello, CFLint";
```
